tests/test_integration.py

```diff
@@ -0,0 +1,127 @@
+            '17:00', '18:00', '19:00', '20:00', '21:00', '22:00', '23:00'],
+            'format': 'netcdf', 'pressure_level': [1, 2, 3, 5, 7, 10, 20, 30,
+            50, 70, 100, 125, 150, 175, 200, 225, 250, 300, 350, 400, 450, 500,
+            550, 600, 650, 700, 750, 775, 800, 825, 850, 875, 900, 925, 950,
+            975, 1000], 'product_type': 'reanalysis', 'day': ['01', '02', '03',
+            '04', '05', '06', '07', '08', '09', '10', '11', '12', '13', '14',
+            '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25',
+            '26', '27', '28', '29', '30', '31']}
+            era5_geopotential_2008_hourly.nc""")
+    },
+    {
+        # preliminary-back-extension is combined with monthly-means
+        "call": dedent("""\
+            era5cli monthly --variables temperature --startyear 1960 --prelimbe
+            --dryrun"""),
+        "result": dedent("""\
+            reanalysis-era5-pressure-levels-monthly-means-preliminary-back-extension
+            {'variable': 'temperature', 'year': 1960, 'month': ['01', '02',
+            '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
+            'time': ['00:00'], 'format': 'netcdf', 'pressure_level': [1, 2, 3,
+            5, 7, 10, 20, 30, 50, 70, 100, 125, 150, 175, 200, 225, 250, 300,
+            350, 400, 450, 500, 550, 600, 650, 700, 750, 775, 800, 825, 850,
+            875, 900, 925, 950, 975, 1000], 'product_type':
+            'reanalysis-monthly-means-of-daily-means'}
+            era5_temperature_1960_monthly.nc""")
+    },
+    {
+        # era5-Land is combined with monthly means
+        "call": dedent("""\
+            era5cli monthly --variables snow_cover --startyear 2008 --land
+            --dryrun"""),
+        "result": dedent("""\
+            reanalysis-era5-land-monthly-means {'variable': 'snow_cover',
+            'year': 2008, 'month': ['01', '02', '03', '04', '05', '06', '07',
+            '08', '09', '10', '11', '12'], 'time': ['00:00'], 'format':
+            'netcdf', 'product_type': 'monthly_averaged_reanalysis'}
+            era5-land_snow_cover_2008_monthly.nc""")
+    }
+]
+
+
+def clean_ids(call):
+    call = call.replace('\n', ' ')
+    call = call.replace('--dryrun', '')
+    return(call)
+
+
+ids = [clean_ids(item["call"]) for item in call_result]
+
+
+@pytest.mark.parametrize("call_result", call_result, ids=ids)
+def test_main(call_result, capsys, caplog):
+    call = call_result["call"].split()
+    result = call_result["result"].replace('\n', ' ') + '\n'
+    # until the actual fetch is monkeypatched, make sure the tests are dryruns
+    if '--dryrun' not in call:
+        pytest.fail('call must be a dryrun')
+    with caplog.at_level(logging.INFO):
+        main(call)
+    captured = capsys.readouterr().out
+    assert result == captured
+    try:
+        warn = call_result["warn"]
+        assert warn in caplog.text
+    except KeyError:
+        pass
```

Comment:
Instead of doing nothing can you check if there is no warning what so ever
```suggestion
        assert caplog.text is ''
```